lib/common/iso8601.c

```diff
@@ -1726,38 +1726,37 @@ pcmk__time_format_hr(const char *format, const pcmk__time_hr_t *hr_dt)
 
 /*!
  * \internal
- * \brief Return human-friendly string corresponding to a time
+ * \brief Return a human-friendly string corresponding to an epoch time value
  *
- * \param[in] when   Pointer to epoch time value (or NULL for current time)
+ * \param[in]  source  Pointer to epoch time value (or \p NULL for current time)
+ * \param[in]  flags   Group of \p crm_time_* flags controlling display format
+ *                     format (0 to use \p ctime() with newline removed)
  *
- * \return Current time as string (as by ctime() but without newline) on
- *         success, NULL otherwise
- * \note The return value points to a statically allocated string which might be
- *       overwritten by subsequent calls to any of the C library date and time
- *       functions.
+ * \return String representation of \p source on success (may be empty depending
+ *         on \p flags); guaranteed not to be \p NULL
+ *
+ * \note The caller is responsible for freeing the return value using \p free().
  */
-const char *
-pcmk__epoch2str(const time_t *when)
+char *
+pcmk__epoch2str(const time_t *source, uint32_t flags)
```

Comment:
I did some work (which I set aside for lack of sufficient justification) toward making `pcmk__epoch2str()` safely return a `const char *`, just like `ctime()`. It would be stored in a static buffer by default, or optionally in a user-specified buffer passed by argument. That buffer would need to be `>= DATE_MAX` to be used safely. `DATE_MAX` would be moved to an internal header as `PCMK__DATE_MAX` for use by callers.

I also did the same for `crm_time_as_string()`, making it a wrapper that `strdup`s an internal function that returns a `const char *`.

If we ever want that, those changes aren't too complicated and make life a little bit easier for callers, who can use the return value directly instead of storing and freeing it. However:
* Efficiency gains are too trivial to matter (output purposes only).
* We can't check the custom buffer size within `pcmk__epoch2str()` if we accept a custom buffer -- we'd have to trust the caller.
* Ownership is an issue... The caller is responsible for passing a custom buffer if there is any chance of something else calling `pcmk__epoch2str()` a second time before the caller is finished using the first return value. If someone forgets to do that and the buffer gets overwritten, that bug will be hard to notice and track down.